test/DynamoCoreWpfTests/WorkspaceGeneralOperations.cs

```diff
@@ -113,7 +113,64 @@ public void VerifyGlobalLacingStrategyUpdatesForLacingStrategies()
             AssertPreviewCount(nodeIds[1].ToString(), 2);
             AssertPreviewCount(nodeIds[2].ToString(), 2);
         }
+        
+        [Test]
+        public void VerifyLacingStrategyForCodeBlockNodes()
+        {
+            var openPath = Path.Combine(TestDirectory, @"core\visualization\LacingStrategyCodeBlockNodes.dyn");
+            ViewModel.OpenCommand.Execute(openPath);
+
+            var workspace = ViewModel.Model.CurrentWorkspace as HomeWorkspaceModel;
+            workspace.RunSettings.RunType = RunType.Automatic;
+
+            Guid codeBlockNodeId = Guid.Parse("5a35517215434699afe122bc51aeff7d");
+            Guid otherNodeId = Guid.Parse("ab8afb7c1dfe4dd0994662f3306fc530");
+            var codeBlockNode = workspace.NodeFromWorkspace(codeBlockNodeId);
+            var otherNode = workspace.NodeFromWorkspace(otherNodeId);
+
+            // Verify initial lacing state is Auto and nodes return correct number of results
+            AssertPreviewCount(codeBlockNodeId.ToString(), 2);
+            AssertPreviewCount(otherNodeId.ToString(), 2);
```

Comment:
Assume this is why you set the RunType to automatic?

Do we actually need to assert that the output of the node is correct? I would assume that this test only cares about checking if the lacing strategy is correct?

Reply:
The existing `VerifyGlobalLacingStrategyUpdatesForLacingStrategies()` test uses `AssertPreviewCount` to test whether the lacing has been changed successfully. I added the `Assert.AreEqual(LacingStrategy.Disabled, codeBlockNode.ArgumentLacing);` when I noted that similarly checked the value had changed successfully but opted the leave the `AssertPreviewCount` checks as well just in case the values for some reason diverged. Should I remove them?

https://github.com/StudioLE/Dynamo/blob/706aaa98488c05abf297891d3cbaeac7e7e89634/test/DynamoCoreWpfTests/WorkspaceGeneralOperations.cs#L85-L90

Reply:
I would say so, as we are already checking that changing lacing will affect the output in `VerifyGlobalLacingStrategyUpdatesForLacingStrategies()`. This test should only test that codeblocks cant have their lacing changed, so maybe name should be something like `VerifyLacingStrategyCantBeChangedOnCodeBlocks()`.

That way we can also remove all `AssertPreviewCount` and the RunType settings

Reply:
Fixed in 3dde853